File: CS/NavigationDemo.Wpf/ViewModels/NavigationPane/MailViewModel.cs
using System.Collections.ObjectModel;
using DevExpress.Mvvm;

namespace NavigationDemo {
    public class MailViewModel : ISupportParameter {
        public virtual ObservableCollection<MailItem> ItemsSource { get; set; }
        public virtual object Parameter { get; set; }

        public virtual int SentGroupIndex { get; set; }
        public virtual int ReceiveGroupIndex { get; set; }

        public MailViewModel() {
            SentGroupIndex = -1;
            ReceiveGroupIndex = -1;
        }
        protected virtual void OnParameterChanged() {
            var id = (NavigationId)Parameter;
            switch(id) {
                case NavigationId.Inbox: SentGroupIndex = -1; ReceiveGroupIndex = 0; break;
                case NavigationId.Outbox: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
                case NavigationId.SentItems: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
                case NavigationId.DeletedItems: SentGroupIndex = -1; ReceiveGroupIndex = 0; break;
                case NavigationId.Drafts: SentGroupIndex = 0; ReceiveGroupIndex = -1; break;
            }
            ItemsSource = NavigationPaneData.MailData[id];
        }
        #region ISupportParameter
        object ISupportParameter.Parameter {
            get { return Parameter; }
            set { Parameter = value; }
        }
        #endregion
    }
}

File: VB/NavigationDemo.Wpf/ViewModels/NavigationPane/MailViewModel.vb
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm

Namespace NavigationDemo
    Public Class MailViewModel
        Implements ISupportParameter

        Public Overridable Property ItemsSource() As ObservableCollection(Of MailItem)
        Public Overridable Property Parameter() As Object

        Public Overridable Property SentGroupIndex() As Integer
        Public Overridable Property ReceiveGroupIndex() As Integer

        Public Sub New()
            SentGroupIndex = -1
            ReceiveGroupIndex = -1
        End Sub
        Protected Overridable Sub OnParameterChanged()
            Dim id = DirectCast(Parameter, NavigationId)
            Select Case id
                Case NavigationId.Inbox
                    SentGroupIndex = -1
                    ReceiveGroupIndex = 0
                Case NavigationId.Outbox
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
                Case NavigationId.SentItems
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
                Case NavigationId.DeletedItems
                    SentGroupIndex = -1
                    ReceiveGroupIndex = 0
                Case NavigationId.Drafts
                    SentGroupIndex = 0
                    ReceiveGroupIndex = -1
            End Select
            ItemsSource = NavigationPaneData.MailData(id)
        End Sub
        #Region "ISupportParameter"
        Private Property ISupportParameter_Parameter() As Object Implements ISupportParameter.Parameter
            Get
                Return Parameter
            End Get
            Set(ByVal value As Object)
                Parameter = value
            End Set
        End Property
        #End Region
    End Class
End Namespace
